### ISSUE TITLE
Issues with Scenes

### ISSUE THREAD

**Issue by TinTin1969** [Breina/ha-artnet-led#62]
Hi All,

First of all, I LOVE this integration as it has finally enabled me to switch on my DMX pool lights with Home Assistant. Previously, I needed a seperate DMX app. So I went and set up a scene, ie. For the pool lights to switch on after sunset and then off again 4hs later, very simple.

However, more times than not, the device switch on Home Assistant shows the light as on after the scene but the light is off. I then have to switch it off and then on manually, which at first makes the light flash, then go off, then go on again. Same happens when the scene finishes, ie the light shows as off but is still on. Interestingly, it doesn't always do it, sometimes it works perfectly. I'm too noob to know where to dig for error logs etc, so how would I go about it to fix this?

Aby help greatly appreciated

**Comment by Breina** [comment]
If you are using a separate ArtNet controller in conjunction with this integration, you'll need to use `node_type: artnet-controller`, which will enable it to receive lighting updates from your DMX app.

Light flashes are usually caused by configuring a wrong bit-width (8 bit / 16 bit ...). Might help to temporarily configure your lights channel by channel and confirm it works (i.e. a light with `type: dimmer`, that only controls the red channel). This helps you narrow down any misconfigured channels.

Logs can be found by going to Settings > System > Logs, then filter on `artnet` and click Load full logs. Then click the download button in the top right and upload them here please.

**Comment by TinTin1969** [comment]
Thank you for your quick reply. I tried switching to Artnet-controller mode and it won't work at all in that mode. It works in artnet-direct mode. 

But to be clear, I don't want to use DMX apps, I only want to use Home Assistant where I have it set up as a light fixture. 

The flashing is just once when I manually change its wrong state to the correct one ie when it shows as on but is actually off, then I need to click the light button twice to bring it to the correct state.

So it works fine by manually clicking the light fixture in Home Assistant, but it doesn't work (often) when using a scene in Home Assistant, e.g. switch on lights after sunset. The light button switches on correctly but not the actual light.

**Comment by TinTin1969** [comment]
Update:

I realised it is not the timer/scene that is the issue. Even when pressing the the device icon manually, sometimes the lights do not react. Meaning I press the light icon on HA and it switches on, but the light doesn't actually switch on. Then I have to press it again (to turn off) and on again and only then the light switches on. Same thing the other way, ie when switching it off. So somehow there is a disconnect between HA and the Artnet extension. How can I troubleshoot this?

**Comment by Breina** [comment]
Yeah I think that's a threading problem. The integration does its thing and then hands back control to HomeAssistant. We ask HomeAssistant to call us back when it's our time to shine again, this can sometimes take a second or two. I haven't found a golden solution to fix this properly (moving part of it to an add-on maybe?).
